knowledge graph/kg_agent.py

```
@@ -0,0 +1,326 @@
+    
+    def __init__(self, uri, user, password):
+        """Initialize the Knowledge Graph with Neo4j connection"""
+        self.driver = neo4j.AsyncGraphDatabase.driver(
+            uri, auth=(user, password)
+        )
+    
+    async def create_entity(self, entity_type: str, properties: Dict[str, Any]):
+        """Create a new entity node in the knowledge graph"""
+        async with self.driver.session() as session:
+            # convert properties to cypher parameters
+            props_string = ", ".join([f"{k}: ${k}" for k in properties.keys()])
```

Ruff (0.12.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)